blockchain/state_transition.go

```diff
@@ -406,7 +409,7 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	}
 
 	// Check whether the init code size has been exceeded.
-	if rules.IsShanghai && msg.To() == nil && len(st.data) > params.MaxInitCodeSize {
+	if rules.IsShanghai && msgTo == nil && len(st.data) > params.MaxInitCodeSize {
```

Comment:
How's about changing like next? It will be clearer.
```
var (
   msg          = st.msg
   msgTo        = msg.To()
   contractCreation = msg.To() == nil
) 
if contractCreation {
   return nil, fmt.Errorf("%w: code size %v limit %v", ErrMaxInitCodeSizeExceeded, len(st.data), params.MaxInitCodeSize)
} else {
   ...
}
```

Reply:
@yoomee1313 
I think so. BTW, do you mind using the not condition because we don't need the opposite condition?
```
if !contractCreation {
   ...
}
```

Reply:
Yes, that works for me. I’m fine with using the not condition instead.

Reply:
@tnasu Could you change the line 413 also?
```
	if rules.IsShanghai && msgTo == nil && len(st.data) > params.MaxInitCodeSize {
```